test/api.test.ts

```diff
@@ -0,0 +1,106 @@
+import assert from 'assert';
+import supertest from 'supertest';
+import app from '../src/app';
+import mongoose from 'mongoose';
+import userModel from '../src/models/userModel';
+
+beforeEach((done) => {
+  mongoose.connect("mongodb://localhost:27017/TestDB",
+  { useNewUrlParser: true, useUnifiedTopology: true },
+  () => done());
+});
+
+afterEach((done) => {
+  mongoose.connection.db.dropDatabase(() => {
+    mongoose.connection.close(() => done())
+  });
+});
+
+const createDbEntry = async () => {
```

Comment:
consider following "factory pattern" for javascript tests. create file "tests/create.ts" and export two functions for each collection in your system:

```
create.userDTO = function(options: Partial<IUser>): IUser { return {}  } // returns javascript object
create.user = function(options: Partial<IUser>): IUser { 
  return userModel.create({ ...create.userDTO(), ...(options || {} });
} // returns javascript object
```